### PR TITLE
Afpacket improvements/v28

### PR DESCRIPTION
**Pull request by victorjulien**
Fix tpacket-v2 implementation dead locking under load.
Remove tpacket-v1 implementation.
Fix tpacket-v3 ifdown/ifup logic.
Change VLAN handling so logging can have the full data
Remove autofp support
Various optimizations and cleanups.

https://redmine.openinfosecfoundation.org/issues/4785
https://redmine.openinfosecfoundation.org/issues/4796
https://redmine.openinfosecfoundation.org/issues/4800

Draft because of https://redmine.openinfosecfoundation.org/issues/4806
If we agree autofp can go, some other commits can be squashed most likely.

Changes since #6574:
- remove autofp
- fix to vlan logic
- minor other fixes and cleanups

### CODE REVIEW

File: src/source-af-packet.c [hunk 1]
```diff
@@ -746,86 +629,44 @@ static TmEcode AFPWritePacket(Packet *p, int version)
...
     /* Need to be in copy mode and need to detect early release
        where Ethernet header could not be set (and pseudo packet) */
-    if ((p->afp_v.copy_mode != AFP_COPY_MODE_NONE) && !PKT_IS_PSEUDOPKT(p)) {
+    if (p->afp_v.copy_mode != AFP_COPY_MODE_NONE) {
         AFPWritePacket(p, TPACKET_V2);
     }
 
-    if (AFPDerefSocket(p->afp_v.mpeer) == 0)
-        goto cleanup;
+    BUG_ON(p->afp_v.relptr == NULL);
```
Comment (jufajardini) [inline src/source-af-packet.c:653]: Was trying to run it locally, got this:
`suricata: source-af-packet.c:653: AFPReleaseDataFromRing: Assertion `!(p->afp_v.relptr == ((void *)0))' failed.
Aborted`

Comment (victorjulien) [reply src/source-af-packet.c:653]: Can you get a back trace?

And in gdb print p->pkt_src?

Comment (jufajardini) [reply src/source-af-packet.c:653]: I've never done that, but I'll try!

Comment (victorjulien) [reply src/source-af-packet.c:653]: Think I see the issue

Comment (victorjulien) [reply src/source-af-packet.c:653]: https://github.com/OISF/suricata/pull/6578/commits/7b0ca4c62dd959dc8d211355c1f3d4f6d216e475

**Comment by victorjulien** [comment]
replaced by #6578

**Pull request closed without merging**